components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -282,6 +284,29 @@ func hasMetadataAccess() bool {
 	return false
 }
 
+func reaper(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	sigs := make(chan os.Signal, 128)
+	signal.Notify(sigs, syscall.SIGCHLD)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-sigs:
+		}
+
+		pid, err := unix.Wait4(-1, nil, 0, nil)
+
+		if err == unix.ECHILD {
+			// The calling process does not have any unwaited-for children.
+			continue
+		}
+
+		log.WithField("pid", pid).Debug("reaped child process")
```

Comment:
Since a zombie process indicates a resource leak that's caused by the parent process not having called `wait()` on the child process, would it make sense to log the parent processes name+pid here, too? 
This would allow us to understand who cause the zombies and do something about it.

Reply:
Unfortunately, by the time we reap the zombie we don't know about the parent anymore (which is the reason we need to reap the processes ourselves).